tests/components/application-connector/test/compass-runtime-agent/synchronisation_test.go

```diff
@@ -65,3 +71,34 @@ func (cs *CompassRuntimeAgentSuite) assignApplicationToFormationAndWaitForSync(a
 		Timeout:              appCreationTimeout,
 	}.Do()
 }
+
+func (cs *CompassRuntimeAgentSuite) removeApplicationAndWaitForSync(appReader ApplicationReader, compassAppName, applicationID string) error {
+	t := cs.T()
+	t.Helper()
+
+	exec := func() error {
+		err := cs.directorClient.UnassignApplication(applicationID, cs.formationName)
+		if err != nil {
+			return err
+		}
+
+		err = cs.directorClient.UnregisterApplication(applicationID)
+		return err
+	}
+
+	verify := func() bool {
+		_, err := appReader.Get(context.Background(), compassAppName, v1.GetOptions{})
+		if err != nil {
+			t.Log(fmt.Sprintf("Failed to get app: %v", err))
+		}
+
+		return err != nil
```

Comment:
Is there an option to check if the error is the `NotFound` error? There could be some another problem like something wrong with the API server or something else